src/ansys/systemcoupling/core/session.py

```diff
@@ -176,10 +176,9 @@ def _get_api_root(self, category):
             version = sycproxy.get_version()
             self.__syc_version = version.replace(".", "_")
         root = get_root(sycproxy, category=category, version=self.__syc_version)
+        self.__part_mgr = ParticipantManager(self, self.__syc_version)
```

Comment:
This is just about there now I think but a small change is needed here. It needs to check that self.__part_mgrt is None before assigning it as the same instance needs to be shared between API roots.